code/modules/mob/living/carbon/carbon.dm

```diff
@@ -616,4 +611,289 @@ var/const/GALOSHES_DONT_HELP = 4
+		human_mob.remove_all_embedded_objects()
+	updatehealth()
+	update_fire()
+	if(dna)
+		for(var/datum/mutation/human/HM in dna.mutations)
+			if(HM.quality != POSITIVE)
+				dna.remove_mutation(HM.name)
+	update_sight()
+	update_vision_overlays()
+	update_canmove()
+
+
+//this handles hud updates
+/mob/living/carbon/update_damage_hud()
+
+	if(!client)
+		return
+
+	if(damageoverlay)
+		damageoverlay.overlays.Cut()
+		switch(stat)
+			if(UNCONSCIOUS)
+				//Critical damage passage overlay
+				if(health <= config.health_threshold_crit)
+					var/image/I = image("icon" = 'icons/mob/screen_full.dmi', "icon_state" = "passage0")
+					I.blend_mode = BLEND_OVERLAY //damageoverlay is BLEND_MULTIPLY
+					switch(health)
+						if(-20 to -10)
+							I.icon_state = "passage1"
+						if(-30 to -20)
+							I.icon_state = "passage2"
+						if(-40 to -30)
+							I.icon_state = "passage3"
+						if(-50 to -40)
+							I.icon_state = "passage4"
+						if(-60 to -50)
+							I.icon_state = "passage5"
+						if(-70 to -60)
+							I.icon_state = "passage6"
+						if(-80 to -70)
+							I.icon_state = "passage7"
+						if(-90 to -80)
+							I.icon_state = "passage8"
+						if(-95 to -90)
+							I.icon_state = "passage9"
+						if(-INFINITY to -95)
+							I.icon_state = "passage10"
+					damageoverlay.overlays += I
+			if(CONSCIOUS)
+				//Oxygen damage overlay
+				if(oxyloss)
+					var/image/I = image("icon" = 'icons/mob/screen_full.dmi', "icon_state" = "oxydamageoverlay0")
+					switch(oxyloss)
+						if(10 to 20)
+							I.icon_state = "oxydamageoverlay1"
+						if(20 to 25)
+							I.icon_state = "oxydamageoverlay2"
+						if(25 to 30)
+							I.icon_state = "oxydamageoverlay3"
+						if(30 to 35)
+							I.icon_state = "oxydamageoverlay4"
+						if(35 to 40)
+							I.icon_state = "oxydamageoverlay5"
+						if(40 to 45)
+							I.icon_state = "oxydamageoverlay6"
+						if(45 to INFINITY)
+							I.icon_state = "oxydamageoverlay7"
+					damageoverlay.overlays += I
+
+				//Fire and Brute damage overlay (BSSR)
+				var/hurtdamage = getBruteLoss() + getFireLoss() + damageoverlaytemp
+				if(hurtdamage)
+					var/image/I = image("icon" = 'icons/mob/screen_full.dmi', "icon_state" = "brutedamageoverlay0")
+					I.blend_mode = BLEND_ADD
+					switch(hurtdamage)
+						if(5 to 15)
+							I.icon_state = "brutedamageoverlay1"
+						if(15 to 30)
+							I.icon_state = "brutedamageoverlay2"
+						if(30 to 45)
+							I.icon_state = "brutedamageoverlay3"
+						if(45 to 70)
+							I.icon_state = "brutedamageoverlay4"
+						if(70 to 85)
+							I.icon_state = "brutedamageoverlay5"
+						if(85 to INFINITY)
+							I.icon_state = "brutedamageoverlay6"
+					var/image/black = image(I.icon, I.icon_state) //BLEND_ADD doesn't let us darken, so this is just to blacken the edge of the screen
+					black.color = "#170000"
+					damageoverlay.overlays += I
+					damageoverlay.overlays += black
+
+/mob/living/carbon/update_health_hud(shown_health_amount)

```

Comment:
the argument is used by humans so the proc uses "health - staminaloss" instead of just "health".